### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by plaindocs**
A bunch of improvements and refactoring, mostly to the Guides

* Getting started
* Validating .travis.yml files
* Customizing Your Build
* Services and Database setup
* Speeding up the build

A number of changes to the _config, including changes to the markdown parser and redirection gem (which did not seem to work for me locally) so use **CAUTION** when merging this in.

Comments welcome here or on slack.

### CODE REVIEW

**Comment by joshk** [comment]
@plaindocs thanks for creating the PR!

Are you able to solve the merge conflicts? or do you need a hand with this?

File: user/getting-started.md [hunk 1]
```diff
@@ -4,92 +4,53 @@ layout: en
...
-* [PHP](/user/languages/php)
-* [Python](/user/languages/python)
-* [Ruby](/user/languages/ruby)
-* [Rust](/user/languages/rust)
-* [Scala](/user/languages/scala)
-* [Visual Basic](/user/languages/csharp/)
+{{ site.data.lists.languages }}
 
+<!-- 
+[C](/user/languages/c) | [C++](/user/languages/cpp) | [Clojure](/user/languages/clojure) | [C#](/user/languages/csharp/) | [D](/user/languages/d) | [Erlang](/user/languages/erlang) | [F#](/user/languages/csharp/) | [Go](/user/languages/go) | [Groovy](/user/languages/groovy) | [Haskell](/user/languages/haskell) | [Java](/user/languages/java) | [JavaScript (with Node.js)](/user/languages/javascript-with-nodejs) | [Julia](/user/languages/julia) | [Objective-C](/user/languages/objective-c) | [Perl](/user/languages/perl) | [PHP](/user/languages/php) | [Python](/user/languages/python) | [Ruby](/user/languages/ruby) | [Rust](/user/languages/rust) | [Scala](/user/languages/scala) | [Visual Basic](/user/languages/csharp/).
```
Comment (BanzaiMan) [inline user/getting-started.md]: Missing: Android and R.

File: _data/lists.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+languages:  |
```
Comment (BanzaiMan) [inline _data/lists.yml]: Android and R a missing from this list.

**Comment by BanzaiMan** [comment]
After conflict resolution and the addition of Android and R, it would look like this: https://github.com/plaindocs/docs-travis-ci-com/compare/gh-pages...travis-ci:ha-test-pr-202?expand=1

**Comment by plaindocs** [comment]
@lislis @joshk could you guys have a look at the landing page (index)?

The obvious outstanding thing is the crappy aligment of the 3 column list. @lislis If that is not fixable, how could we make it better? The UL is generated from markdown, and has the language-docs class. Otherwise I think it looks ok to me.

**Pull request merged**